### PR TITLE
updated game model and controller with scope method

### PR DESCRIPTION
**Pull request by zaynrahn**
Recreated new branch to take care of merge conflicts in previous branch. @jacobwgillespie

### CODE REVIEW

**Comment by abdullahalam** [comment]
Looks good to me and good job @zaynrahn

File: app/models/game.rb [hunk 1]
```diff
@@ -1,5 +1,9 @@
 class Game < ApplicationRecord
-  belongs_to :white_player, class_name: 'Player'
-  belongs_to :black_player, class_name: 'Player'
+  belongs_to :white_player, class_name: 'User'
+  belongs_to :black_player, class_name: 'User', required: false
   has_many :pieces
+
+  def self.needs_second_player?
```
Comment (jacobwgillespie) [inline app/models/game.rb]: One small suggestion - this method will return a listing of all games that match the `where` condition, not true or false, so you may want to remove the `?` from the method name.

Comment (zaynrahn) [inline app/models/game.rb]: Will take care of it. Thanks!

**Comment by zaynrahn** [comment]
@abdullahalam Thank you!

**Pull request merged**